File: Microsoft.Recognizers.Text.DateTime/Parsers/IHolidayParserConfiguration.cs
using System;
using System.Collections.Generic;
using System.Collections.Immutable;
using System.Text.RegularExpressions;
using DateObject = System.DateTime;

namespace Microsoft.Recognizers.Text.DateTime.Parsers
{
    public interface IHolidayParserConfiguration
    {
        IImmutableDictionary<string, string> VariableHolidaysTimexDictionary { get; }
        IImmutableDictionary<string, Func<int, DateObject>> HolidayFuncDictionary { get; }

        IImmutableDictionary<string, IEnumerable<string>> HolidayNames { get; }
        
        IEnumerable<Regex> HolidayRegexList { get; }

        int GetSwiftYear(string text);

        string SanitizeHolidayToken(string holiday);
    }
}

File: Java/libraries/recognizers-text-date-time/src/main/java/com/microsoft/recognizers/text/datetime/parsers/config/IHolidayParserConfiguration.java
package com.microsoft.recognizers.text.datetime.parsers.config;

import com.google.common.collect.ImmutableMap;
import com.microsoft.recognizers.text.datetime.config.IOptionsConfiguration;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.function.Function;
import java.util.function.IntFunction;
import java.util.regex.Pattern;

public interface IHolidayParserConfiguration extends IOptionsConfiguration {
    ImmutableMap<String, String> getVariableHolidaysTimexDictionary();

    ImmutableMap<String, IntFunction<LocalDateTime>> getHolidayFuncDictionary();

    ImmutableMap<String, Iterable<String>> getHolidayNames();

    Iterable<Pattern> getHolidayRegexList();

    int getSwiftYear(String text);

    String sanitizeHolidayToken(String holiday);
}
